Model/ContentRepository.php

```diff
@@ -224,18 +221,24 @@ public function findAnonymouslyCreatedById($id)
             ->andWhere('c.active = :active')
             //->andWhere('c.author IS NULL')
             ->setParameters(['id' => $id, 'active' => 0])
-            ->getQuery()
-        ;
+            ->getQuery();
 
-        return $query->getSingleResult();
+        return $query->useResultCache(true, self::CACHE_TTL)->getSingleResult();
     }
 
     /**
+     * Find popular content items by template
+     *
      * @todo just retrieves random content items for now
+     *
+     * @param     $template
+     * @param int $limit
+     *
+     * @return \Doctrine\Common\Collections\ArrayCollection
      */
-    public function findPopularByTemplate($template)
+    public function findPopularByTemplate($template, $limit = 10)
```

Comment:
While you're at it, could you mark this method as deprecated?

Something like

    @deprecated will be removed in 1.0